src/nimony/sigmatch.nim

```diff
@@ -662,6 +662,27 @@ proc isStringType*(a: Cursor): bool {.inline.} =
 proc isSomeStringType*(a: Cursor): bool {.inline.} =
   result = a.typeKind == CstringT or isStringType(a)
 
+proc isSomeSeqType*(a: Cursor, elemType: var Cursor): bool =
+  # check that `a` is either an instantiation of seq or an invocation to it
+  result = false
+  var a = a
+  if a.kind == Symbol:
```

Comment:
I think you need a while loop here plus a counter due to the possibility of: 

```nim
type
  MyAlias = OtherAlias
  OtherAlias = StupidAlias
  StupidAlias = seq
```

Reply:
Aliases are supposed to be inlined, as in `MyAlias` would immediately resolve to `seq` when semchecked, including with generics, but could still add it for safety